api/services/PokemonHandler.js

```diff
@@ -152,6 +158,13 @@ exports.createPokemonFromPk6 = async ({user, visibility, boxId, file}) => {
   parsed._boxVisibility = box.visibility;
   parsed.owner = user.name;
   parsed.visibility = visibility;
+  parsed.gen = parsed.gen === undefined ? gen : parsed.gen;
```

Comment:
You could simplify this to "parsed.gen = parsed.gen || gen" because parsed.gen will never legitimately be 0.